pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -1194,3 +1221,12 @@ func envs(envs []corev1.EnvVar, envVar corev1.EnvVar) []corev1.EnvVar {
 	}
 	return envs
 }
+
+// Volume names must be valid RFC 1123 Label Names: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+func secretNameToVolumeName(secretName string) (string, error) {
+	volumeName := fmt.Sprintf(volumeSecretNameTemplate, secretName)
+	if len(volumeName) > 63 {
+		return "", errors.Errorf("secret name %s is too long, it must contain fewer than 50 characters", secretName)
```

Comment:
This effectively prevents long secrets from being used because we are enforcing the volume name to contain the secret name. That feels unnecessary? Why not generate a random name or sequence of volume names volume-1,volume-2,...?

Reply:
oh right volume names don't matter across builds, that's what I was concerned about

Reply:
I'm in favor of generating completely random volume names